Automattic/wp-calypso
# Stats: Updating Odyssey docs links

Related to https://github.com/Automattic/wp-calypso/issues/87802

## Proposed Changes

* updating Odyssey links to point to jetpack.com documentation

## Why are these changes being made?


* unifying user experience with the platform they access the product from

## Testing Instructions



* open the live branch
* verify that the inline links point to a proper source of documentation for .com and Jetpack

## Pre-merge Checklist
